Library/Homebrew/utils/pypi.rb

```diff
@@ -225,7 +225,7 @@ def update_python_resources!(formula, version: nil, package_name: nil, extra_pac
 
     # Remove extra packages that may be included in pipgrip output
     exclude_list = %W[#{main_package.name} argparse pip setuptools wheel wsgiref].map { |p| Package.new p }
-    found_packages.delete_if { |package| exclude_list.include? package }
+    found_packages.delete_if { |package| exclude_list.include?(package) && extra_packages.exclude?(package) }
```

Comment:
I've caught myself in trouble trying to understand a line with `exclude..include?..exclude?`😄  
Maybe we could merge `exclude_list` with `exclude_packages` in the top (and apply new logic there), instead of having several types of excluding in several places?
https://github.com/Homebrew/brew/blob/5286be91cbdc34af8276143df3e005f1a86f3d4b/Library/Homebrew/utils/pypi.rb#L183-L184

Reply:
Yeah the current phrasing is definitely a bit awkward. I just pushed a new version which should be a little clearer.